pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
-	var err error
-	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
-		cfg, err = readDockerConfigJson(dockercfgPath)
-	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
-		cfg, err = readDockercfg(dockercfgPath)
-	}
-
+	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
+	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
```

Comment:
my thought would be to introduce a version of the read function that takes a path, and then have the existing flow that uses the preferred path, call that function.  and we could then leverage the new read function from here directly.